packit_service/service/api/webhooks.py

```diff
@@ -206,28 +208,94 @@ def post(self):
 
         return "Webhook accepted. We thank you, Gitlab.", HTTPStatus.ACCEPTED
 
-    @staticmethod
-    def validate_token():
+    def create_confidential_issue_with_token(self):
+        project_data = request.json["project"]
+
+        namespace, repo_name = project_data["path_with_namespace"].split("/")
+        http_url = project_data["git_http_url"]
+
+        project_token = ProjectTokenModel.get_project(
+            namespace=namespace, repo_name=repo_name, project_url=http_url
+        )
+
+        if not project_token:
+            token = jwt.encode(
+                {"namespace": namespace, "repo_name": repo_name},
+                config.gitlab_token_secret,
+                algorithm="HS256",
+            )
+
+            ProjectTokenModel.create(
+                token=token,
+                namespace=namespace,
+                repo_name=repo_name,
+                project_url=http_url,
+            )
+
+            project = config.get_project(url=http_url)
+            packit_user = project.service.user.get_username()
+
+            logger.info("Confidential issue created.")
+
+            project.create_issue(
+                title="Packit-Service Authentication",
+                body=f"To configure Packit-Service with `{repo_name}` you need to\n"
+                f"configure the webhook settings. Head to {http_url}/hooks and add\n"
+                f"the Secret Token `{token}` to authenticate requests coming to Packit.\n\n"
+                f"Packit needs rights to set commit status to merge requests, Please grant\n"
+                f"[{packit_user}](https://gitlab.com/{packit_user}) `admin` permissions\n"
+                f"on the {namespace}/{repo_name} project. You can add the rights by clicking\n"
+                f"[here]({http_url}/-/project_members).",
```

Comment:
This is probably unnecessary here -- since we are creating confidential issues we already have that permission..;)

(We might want to create a regular issue if we don't have that permission.)

Reply:
Anyone with or without permission to the repository can create confidential issues. User which don't have permission cannot see other confidential issues created.

Reply:
Aha, ok. My bad. That does not make sense..;)

Thanks for the explanation.